pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -439,21 +439,33 @@ func newInboundPassthroughFilterChains(env *model.Environment, node *model.Proxy
 			},
 		}
 		setAccessLog(env, node, tcpProxy)
-		filter := &listener.Filter{
+		tcpProxyFilter := &listener.Filter{
 			Name: xdsutil.TCPProxy,
 		}
 
-		if util.IsXDSMarshalingToAnyEnabled(node) {
-			filter.ConfigType = &listener.Filter_TypedConfig{TypedConfig: util.MessageToAny(tcpProxy)}
+		isXDSMarshalingToAnyEnabled := util.IsXDSMarshalingToAnyEnabled(node)
+		if isXDSMarshalingToAnyEnabled {
+			tcpProxyFilter.ConfigType = &listener.Filter_TypedConfig{TypedConfig: util.MessageToAny(tcpProxy)}
 		} else {
-			filter.ConfigType = &listener.Filter_Config{Config: util.MessageToStruct(tcpProxy)}
+			tcpProxyFilter.ConfigType = &listener.Filter_Config{Config: util.MessageToStruct(tcpProxy)}
 		}
 		filterChain := &listener.FilterChain{
 			FilterChainMatch: &filterChainMatch,
 			Filters: []*listener.Filter{
-				filter,
+				tcpProxyFilter,
 			},
 		}
+		// Insert the RBAC filter to the passthrough filter chain.
+		if b := builder.NewBuilder(env.Mesh.TrustDomain, env.Mesh.TrustDomainAliases, nil,
+			node.WorkloadLabels, node.ConfigNamespace, push.AuthzPolicies, isXDSMarshalingToAnyEnabled); b != nil {
+			if authzFilter := b.BuildTCPFilter(); authzFilter != nil {
+				filterChain.Filters = []*listener.Filter{
```

Comment:
nit: append instead of literal slice

Reply:
The authzFilter needs to be added in the beginning of the slice, do you mean something like the following?
```
filterChain.Filters = append([]*listener.Filter{authzFilter}, filterChain.Filters...)
```